### PR TITLE
Add count operator

### PR DESCRIPTION
**Pull request by polonez**
#43 
Add count operator, tests and playground demo.

### CODE REVIEW

**Comment by cruisediary** [comment]
@polonez Please update changelog about your PR 👻

File: Readme.md [hunk 1]
```diff
@@ -582,6 +583,21 @@ next((Test Class, 13))
 completed
 ```
 
+#### count
+
+The `count` operator counts the number of items emitted by an Observable. If predicate exists, then counts the number of items satisfying it.
```
Comment (freak4pc) [inline Readme.md]: Please change this to : 

> Emit the number of items emitted by an Observable once it terminates with no errors. If a predicate is given, only elements matching the predicate will be counted. 

Also, reference the official ReactiveX page: http://reactivex.io/documentation/operators/count.html

File: Readme.md [hunk 2]
```diff
@@ -582,6 +583,21 @@ next((Test Class, 13))
...
+
+```swift
+Observable.from([1, 2, 3, 4, 5, 6])
+    .count { $0 % 2 == 0}
```
Comment (freak4pc) [inline Readme.md]:
```suggestion
    .count { $0 % 2 == 0 }
```

Space before closing curly brace

File: Source/RxSwift/count.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import RxSwift
+
+extension Observable {
+    /**
+     Count the number of items emitted by an Observable
+     - seealso: [count operator on reactivex.io](http://reactivex.io/documentation/operators/count.html)
+     - returns: An Observable sequence containing a value that represents how many elements in the specified observable sequence satisfy a condition if provided, else the count of items.
+     */
+    public func count() -> Observable<Int> {
+        return reduce(0, accumulator: { cnt, _ in cnt + 1 })
```
Comment (freak4pc) [inline Source/RxSwift/count.swift]: Let's get rid of the label and use the trailing-closure syntax here: 

```suggestion
        return reduce(0) { count, _ in count + 1 }
```

File: Playground/RxSwiftExtPlayground.playground/Pages/count.xcplaygroundpage/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+ - returns: An Observable sequence containing a value that represents how many elements in the specified observable sequence satisfy a condition if provided, else the count of items.
+
+ */
+
+example("count") {
+
+    // count even number in the sequence
+    let _ = Observable.from([1...10])
+        .count { $0 % 2 == 0}
```
Comment (freak4pc) [inline Playground/RxSwiftExtPlayground.playground/Pages/count.xcplaygroundpage/Contents.swift]:
```suggestion
        .count { $0 % 2 == 0 }
```

Spacing here as well

File: Playground/RxSwiftExtPlayground.playground/Pages/count.xcplaygroundpage/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        .subscribe(onNext: {
+            print ("\($0)")
```
Comment (freak4pc) [inline Playground/RxSwiftExtPlayground.playground/Pages/count.xcplaygroundpage/Contents.swift]: No need for the casting, simply:

```suggestion
            print($0)
```

**Comment by polonez** [comment]
thank you 👍

**Pull request merged**